mesonbuild/scripts/clippy.py

```diff
@@ -0,0 +1,67 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2024 The Meson development team
+
+from __future__ import annotations
+from collections import defaultdict
+import os
+import tempfile
+import typing as T
+
+from .run_tool import run_tool_on_targets, run_with_buffered_output
+from .. import build, mlog
+from ..mesonlib import MachineChoice, PerMachine
+
+if T.TYPE_CHECKING:
+    from ..compilers.rust import RustCompiler
+
+class ClippyDriver:
+    def __init__(self, build: build.Build, tempdir: str):
+        self.tools: PerMachine[T.List[str]] = PerMachine([], [])
+        self.warned: T.DefaultDict[str, bool] = defaultdict(lambda: False)
+        self.tempdir = tempdir
+        for machine in MachineChoice:
+            compilers = build.environment.coredata.compilers[machine]
+            if 'rust' in compilers:
+                compiler = T.cast('RustCompiler', compilers['rust'])
+                self.tools[machine] = compiler.get_rust_tool('clippy-driver', build.environment)
+
+    def warn_missing_clippy(self, machine: str) -> None:
+        if self.warned[machine]:
+            return
+        mlog.warning(f'clippy-driver not found for {machine} machine')
+        self.warned[machine] = True
+
+    def __call__(self, target: T.Dict[str, T.Any]) -> T.Iterable[T.Coroutine[T.Any, T.Any, int]]:
```

Comment:
Pretty sure that Coroutine is like Generator, so `[None, None, int]`

Reply:
Yep, that works (I fixed it in one of the commits but not this one).